src/App.tsx

```diff
@@ -1,41 +1,107 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { Post, User } from './react-app-env';
+import { getUserPosts, getUsers } from './api/posts';
+import { Loader } from './components/Loader';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsLoaded, setPostsLoaded] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+
+  const loadUsers = useCallback(
+    async () => {
+      const loadedUsers = await getUsers();
+
+      setUsers(loadedUsers);
+    },
+    [],
+  );
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  const firstTenUsers = users.filter(user => user.id <= 10 && user.id >= 1);
+
+  const loadPosts = useCallback(
+    async () => {
+      const loadedPosts = await getUserPosts(selectedUserId);
+
+      setPostsLoaded(true);
+
+      setPosts(loadedPosts);
+    },
+    [],
+  );
```

Comment:
When changing a `user` from the selection the list of posts doesn't change, because you're using an empty dependency array it should have `posts`